tuning/trainercontroller/operations/hfcontrols.py

```diff
@@ -0,0 +1,39 @@
+# Standard
+import re
+import inspect
+from dataclasses import fields
+
+# Third Part
+from transformers import TrainerControl
+
+# Local
+from tuning.trainercontroller.operations import Operation
+
+class HFControls(Operation):
```

Comment:
@alex-jw-brooks The intent of having a layer of abstraction for `custom operations` is as follows:
1. In future, this will help expand the framework beyond huggingface TrainerControl to other usecases (e.g. notifications to end user on training anomalies).
2. It allows for a registration mechanism which helps developers add custom operations without changing the `trainercontroller framework` code.